source/common/tls/ssl_socket.cc

```diff
@@ -377,6 +377,13 @@ void SslSocket::onAsynchronousCertValidationComplete() {
   }
 }
 
+void SslSocket::onAsynchronousCertSelectionComplete() {
+  ENVOY_CONN_LOG(debug, "Async cert selection completed", callbacks_->connection());
+  if (info_->state() == Ssl::SocketState::HandshakeInProgress) {
```

Comment:
Is it possible for this condition to not be true? If not, how does that happen?

Reply:
it follows `onAsynchronousCertSelectionComplete`, I'm not sure about how does that happens, maybe just for in case?